tests/contrib/molten/test_molten.py

```diff
@@ -0,0 +1,171 @@
+    if headers:
+        return client.request('GET', uri, headers=headers)
+    return client.get(uri)
+
+
+class TestMolten(TestCase):
+    """"Ensures Molten is properly instrumented."""
+
+    TEST_SERVICE = 'molten-patch'
+
+    def setUp(self):
+        patch()
+        self.tracer = get_dummy_tracer()
+        Pin.override(molten, tracer=self.tracer)
+
+    def tearDown(self):
+        unpatch()
+        self.tracer.writer.pop()
+        delattr(self, 'tracer')
+
+    def test_route_success(self):
+        """ Tests request was a success with the expected span tags """
+        response = molten_client()
+        spans = self.tracer.writer.pop()
+        self.assertEqual(response.status_code, 200)
+        # TestResponse from TestClient is wrapper around Response so we must
+        # access data property
+        self.assertEqual(response.data, '"Hello 24 year old named Jim!"')
+        span = spans[0]
+        self.assertEqual(span.service, 'molten')
+        self.assertEqual(span.name, 'molten.request')
+        self.assertEqual(span.get_tag('http.method'), 'GET')
+        self.assertEqual(span.get_tag('http.url'), '/hello/Jim/24')
+        self.assertEqual(span.get_tag('http.status_code'), '200')
+
+        # test override of service name
+        Pin.override(molten, service=self.TEST_SERVICE)
+        response = molten_client()
+        spans = self.tracer.writer.pop()
+        self.assertEqual(spans[0].service, 'molten-patch')
+
+    def test_resources(self):
+        """ Tests request has expected span resources """
+        response = molten_client()
+        spans = self.tracer.writer.pop()
+
+        # `can_handle_parameter` appears twice since two parameters are in request
+        # TODO[tahir]: missing ``resolve` method for components
+
+        expected = [
+            'GET /hello/{name}/{age}',
+            'molten.middleware.ResponseRendererMiddleware',
+            'molten.components.HeaderComponent.can_handle_parameter',
+            'molten.components.CookiesComponent.can_handle_parameter',
+            'molten.components.QueryParamComponent.can_handle_parameter',
+            'molten.components.RequestBodyComponent.can_handle_parameter',
+            'molten.components.RequestDataComponent.can_handle_parameter',
+            'molten.components.SchemaComponent.can_handle_parameter',
+            'molten.components.UploadedFileComponent.can_handle_parameter',
+            'molten.components.HeaderComponent.can_handle_parameter',
+            'molten.components.CookiesComponent.can_handle_parameter',
+            'molten.components.QueryParamComponent.can_handle_parameter',
+            'molten.components.RequestBodyComponent.can_handle_parameter',
+            'molten.components.RequestDataComponent.can_handle_parameter',
+            'molten.components.SchemaComponent.can_handle_parameter',
+            'molten.components.UploadedFileComponent.can_handle_parameter',
+            'GET /hello/{name}/{age}',
```

Comment:
I am 50/50 on whether this should be this or the `func_name(wrapped)`.

Reply:
Went with `func_name`